lib/vocabularies/applicator/patternProperties.ts

```diff
@@ -12,9 +13,16 @@ const def: CodeKeywordDefinition = {
   code(cxt: KeywordCxt) {
     const {gen, schema, data, parentSchema, it} = cxt
     const {opts} = it
-    const patterns = schemaProperties(it, schema)
-    // TODO mark properties matching patterns with always valid schemas as evaluated
-    if (patterns.length === 0) return
+    const patterns = allSchemaProperties(schema)
+    const validPatterns = patterns.filter((p) => !alwaysValidSchema(it, schema[p] as AnySchema))
```

Comment:
Multiple negation is confusing. `validPatterns` contains the patterns whose schemas are NOT always valid...

Reply:
True, I agree it may lead to slight confusion.